generator/routing_generator.cpp

```diff
@@ -420,20 +420,20 @@ void BuildRoutingIndex(string const & baseDir, string const & countryName, strin
       int ind2 = -1;
       double dist1 = numeric_limits<double>::max();
       double dist2 = numeric_limits<double>::max();
-      for (int j = 0; j < ft.GetPointsCount(); ++j)
+      for (size_t j = 0; j < ft.GetPointsCount(); ++j)
       {
         double lon = MercatorBounds::XToLon(ft.GetPoint(j).x);
         double lat = MercatorBounds::YToLat(ft.GetPoint(j).y);
         double const d1 = ms::DistanceOnEarth(pts[0].y, pts[0].x, lat, lon);
         double const d2 = ms::DistanceOnEarth(pts[1].y, pts[1].x, lat, lon);
         if (d1 < dist1)
         {
-          ind1 = j;
+          ind1 = static_cast<int>(j);
```

Comment:
dist1 and dist2 will be set to infinity, so invalid cases are easily captured by eye. I'm not insisting, but this is generally not a good practice to mix unsigned variables with signed ones.